src/modm/driver/encoder/encoder_input.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <modm/platform.hpp>
+
+namespace modm
+{
+/**
+ * @brief 				Gray code decoder using STM32 Timer mode
+ *
+ * @tparam Timer 		STM32 General purpose timer
+ * @tparam SignalA		First modm::platform::Gpio pin that connects to Timer::Ch1
+ * @tparam SignalB		Second modm::platform::Gpio pin that connects to Timer::Ch2
+ * @tparam PRESCALER	How many encoder pulses to count as one
+ * @tparam DeltaType	Fast encoders may require int16_t
+ */
+template<class Timer, typename SignalA, typename SignalB, uint16_t PRESCALER = 4,
+	std::signed_integral DeltaType = int8_t>
```

Comment:
```suggestion
	std::signed_integral DeltaType = int16_t>
```
I'd make this 16-bit since saving 1B is probably not that relevant on STM32 and overflow problems are really annoying to debug.